### PR TITLE
Fix crash due to probable race condition

### LINKED ISSUE #12453: V3 scheduler crash while resuming / updating deck counts
###### Reproduction Steps

1. Unknown exactly - but logcat looks pretty robust, can perhaps fish out the scenario from there?
2. 
3. 

###### Expected Result

no crash

###### Actual Result

crash

###### Debug info

https://ankidroid.org/acra/app/1/bug/11281/report/b0298c9f-9f03-4511-87b8-2258d1cce41a

```
Stacktrace (No mapping found)
java.lang.RuntimeException: Unable to resume activity {com.ichi2.anki.a/com.ichi2.anki.Reviewer}: java.util.NoSuchElementException: List is empty.
	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4631)
	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4663)
	at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:52)
	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2246)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:233)
	at android.app.ActivityThread.main(ActivityThread.java:8010)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:631)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:978)
Caused by: java.util.NoSuchElementException: List is empty.
	at kotlin.collections.CollectionsKt___CollectionsKt.first(_Collections.kt:8)
	at kotlin.collections.CollectionsKt.first(Unknown Source:0)
	at com.ichi2.libanki.sched.SchedV3.countIdx(SchedV3.kt:2)
	at com.ichi2.anki.Reviewer.updateScreenCounts(Reviewer.kt:13)
	at com.ichi2.anki.Reviewer.updateActionBar(Reviewer.kt:2)
	at com.ichi2.anki.AbstractFlashcardViewer.onResume(AbstractFlashcardViewer.kt:5)
	at com.ichi2.anki.Reviewer.onResume(Reviewer.kt:2)
	at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1571)
	at android.app.Activity.performResume(Activity.java:8141)
	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4621)
```

logcat

```

09-18 23:51:04.095 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.widgets.DeckAdapter.getEta(DeckAdapter.kt:2)
09-18 23:51:04.095 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: kotlin.io.CloseableKt.closeFinally(Closeable.kt:1)
09-18 23:51:04.096 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.widgets.DeckAdapter.getEta(DeckAdapter.kt:2)
09-18 23:51:04.098 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.widgets.DeckAdapter.getEta(DeckAdapter.kt:2)
09-18 23:51:04.098 W/AnkiDroid(16934): Backend$checkMainThreadOp/ select avg(case when type = 0 then case when ease > 1 then 1.0 else 0.0 end else null end) as newRate, avg(case when type = 0 then time else null end) as newTime, avg(case when type in (1, 3) then case when ease > 1 then 1.0 else 0.0 end else null end) as revRate, avg(case when type in (1, 3) then time else null end) as revTime, avg(case when type = 2 then case when ease > 1 then 1.0 else 0.0 end else null end) as relrnRate, avg(case when type = 2 then time else null end) as relrnTime from revlog where id > ?
09-18 23:51:04.099 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.widgets.DeckAdapter.getEta(DeckAdapter.kt:2)
09-18 23:51:04.100 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.widgets.DeckAdapter.getEta(DeckAdapter.kt:2)
09-18 23:51:04.100 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: kotlin.io.CloseableKt.closeFinally(Closeable.kt:1)
09-18 23:51:04.101 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.DeckPicker.renderPage(DeckPicker.kt:24)
09-18 23:51:04.102 W/AnkiDroid(16934): Backend$checkMainThreadOp/ SELECT count() FROM cards
09-18 23:51:04.102 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.DeckPicker.renderPage(DeckPicker.kt:24)
09-18 23:51:04.102 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.DeckPicker.renderPage(DeckPicker.kt:24)
09-18 23:51:04.103 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.DeckPicker.renderPage(DeckPicker.kt:35)
09-18 23:51:04.104 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.DeckPicker.renderPage(DeckPicker.kt:35)
09-18 23:51:06.671 I/AnkiDroid(16934): AbstractFlashcardViewer:: Show answer button pressed
09-18 23:51:06.671 I/AnkiDroid(16934): stop: automatically show answer
09-18 23:51:06.683 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.Reviewer.displayAnswerBottomBar$nextIvlStr(Unknown Source:14)
09-18 23:51:06.695 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.Reviewer.displayAnswerBottomBar$nextIvlStr(Unknown Source:14)
--------- beginning of perf
09-18 23:51:07.753 I/AnkiDroid(16934): stop: automatically show question
09-18 23:51:07.753 I/AnkiDroid(16934): AbstractFlashcardViewer:: EASE_3 pressed
09-18 23:51:07.758 I/AnkiDroid(16934): Answering card 1607617819496
09-18 23:51:07.786 I/AnkiDroid(16934): Obtaining next card
09-18 23:51:07.791 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.AbstractFlashcardViewer$NextCardHandler.dealWithTimeBox(AbstractFlashcardViewer.kt:2)
09-18 23:51:07.794 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.cardviewer.SoundPlayer$CardSoundConfig$Companion.create(SoundPlayer.kt:1)
09-18 23:51:07.794 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.cardviewer.SoundPlayer$CardSoundConfig$Companion.create(SoundPlayer.kt:1)
09-18 23:51:07.796 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.reviewer.AnswerTimer.setupForCard(AnswerTimer.kt:2)
09-18 23:51:07.797 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.reviewer.AnswerTimer.setupForCard(AnswerTimer.kt:2)
09-18 23:51:07.798 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.reviewer.AnswerTimer.resetTimerUI(AnswerTimer.kt:7)
09-18 23:51:07.799 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.reviewer.AnswerTimer.resetTimerUI(AnswerTimer.kt:7)
09-18 23:51:07.799 I/AnkiDroid(16934): ViewGroupUtils::setRenderWorkaround - using default / hardware rendering
09-18 23:51:07.803 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.AbstractFlashcardViewer.updateDeckName(AbstractFlashcardViewer.kt:3)
09-18 23:51:07.805 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.AbstractFlashcardViewer.updateDeckName(AbstractFlashcardViewer.kt:3)
09-18 23:51:07.806 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.Reviewer.updateScreenCounts(Reviewer.kt:4)
09-18 23:51:07.809 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.Reviewer.updateScreenCounts(Reviewer.kt:13)
09-18 23:51:07.811 I/AnkiDroid(16934): AbstractFlashcardViewer:: Question successfully shown for card id 1607617819493
09-18 23:51:07.819 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.Reviewer.onCreateOptionsMenu(Reviewer.kt:26)
09-18 23:51:07.820 I/AnkiDroid(16934): undo: UndoStatus(undo=Karte beantworten, redo=null, lastStep=5), false
09-18 23:51:07.824 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.Reviewer.onCreateOptionsMenu(Reviewer.kt:33)
09-18 23:51:07.826 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.AbstractFlashcardViewer.getParentDid(Unknown Source:8)
09-18 23:51:07.827 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.Reviewer.onCreateOptionsMenu(Reviewer.kt:65)
09-18 23:51:07.829 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.Reviewer.suspendNoteAvailable(Reviewer.kt:4)
09-18 23:51:07.830 W/AnkiDroid(16934): Backend$checkMainThreadOp/ select 1 from cards where nid = ? and id != ? and queue != -1 limit 1
09-18 23:51:07.830 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.Reviewer.suspendNoteAvailable(Reviewer.kt:4)
09-18 23:51:07.831 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.Reviewer.suspendNoteAvailable(Reviewer.kt:4)
09-18 23:51:07.832 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.Reviewer.buryNoteAvailable(Reviewer.kt:4)
09-18 23:51:07.832 W/AnkiDroid(16934): Backend$checkMainThreadOp/ select 1 from cards where nid = ? and id != ? and queue >=  0 limit 1
09-18 23:51:07.832 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.Reviewer.buryNoteAvailable(Reviewer.kt:4)
09-18 23:51:07.833 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.Reviewer.buryNoteAvailable(Reviewer.kt:4)
09-18 23:51:09.242 I/AnkiDroid(16934): AbstractFlashcardViewer:: Show answer button pressed
09-18 23:51:09.242 I/AnkiDroid(16934): stop: automatically show answer
09-18 23:51:09.254 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.Reviewer.displayAnswerBottomBar$nextIvlStr(Unknown Source:14)
09-18 23:51:09.267 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.Reviewer.displayAnswerBottomBar$nextIvlStr(Unknown Source:14)
09-18 23:51:10.186 I/AnkiDroid(16934): stop: automatically show question
09-18 23:51:10.186 I/AnkiDroid(16934): AbstractFlashcardViewer:: EASE_3 pressed
09-18 23:51:10.189 I/AnkiDroid(16934): Answering card 1607617819493
09-18 23:51:10.199 I/AnkiDroid(16934): Obtaining next card
09-18 23:51:10.201 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.AbstractFlashcardViewer$NextCardHandler.dealWithTimeBox(AbstractFlashcardViewer.kt:2)
09-18 23:51:10.202 I/AnkiDroid(16934): stop: automatically show answer
09-18 23:51:10.202 I/AnkiDroid(16934): stop: automatically show question
09-18 23:51:10.203 I/AnkiDroid(16934): finishWithAnimation END
09-18 23:51:10.221 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.Reviewer.onCreateOptionsMenu(Reviewer.kt:8)
09-18 23:51:10.221 W/AnkiDroid(16934): Backend$checkMainThreadOp/ SELECT guid, mid, mod, usn, tags, flds, flags, data FROM notes WHERE id = ?
09-18 23:51:10.222 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.Reviewer.onCreateOptionsMenu(Reviewer.kt:8)
09-18 23:51:10.223 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.Reviewer.onCreateOptionsMenu(Reviewer.kt:8)
09-18 23:51:10.223 W/AnkiDroid(16934): Backend$checkMainThreadOp/ select scm from col
09-18 23:51:10.223 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.Reviewer.onCreateOptionsMenu(Reviewer.kt:8)
09-18 23:51:10.224 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: kotlin.io.CloseableKt.closeFinally(Closeable.kt:1)
09-18 23:51:10.224 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: kotlin.io.CloseableKt.closeFinally(Closeable.kt:1)
09-18 23:51:10.226 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.Reviewer.onCreateOptionsMenu(Reviewer.kt:26)
09-18 23:51:10.226 I/AnkiDroid(16934): undo: UndoStatus(undo=Karte beantworten, redo=null, lastStep=6), false
09-18 23:51:10.228 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.Reviewer.onCreateOptionsMenu(Reviewer.kt:33)
09-18 23:51:10.231 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.AbstractFlashcardViewer.getParentDid(Unknown Source:8)
09-18 23:51:10.231 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.Reviewer.onCreateOptionsMenu(Reviewer.kt:65)
09-18 23:51:10.239 I/AnkiDroid(16934): AnkiActivity::onPause - Reviewer
09-18 23:51:10.239 I/AnkiDroid(16934): stop: automatically show answer
09-18 23:51:10.239 I/AnkiDroid(16934): stop: automatically show question
09-18 23:51:10.260 I/AnkiDroid(16934): AnkiActivity::onStart - Reviewer
09-18 23:51:10.263 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.reviewer.AnswerTimer.resume(AnswerTimer.kt:4)
09-18 23:51:10.263 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.reviewer.AnswerTimer.resume(AnswerTimer.kt:4)
09-18 23:51:10.264 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.reviewer.AnswerTimer.resume(AnswerTimer.kt:5)
09-18 23:51:10.265 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.reviewer.AnswerTimer.resume(AnswerTimer.kt:5)
09-18 23:51:10.266 I/AnkiDroid(16934): AnkiActivity::onResume - Reviewer
09-18 23:51:10.269 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.Reviewer.setTitle(Reviewer.kt:2)
09-18 23:51:10.270 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.Reviewer.setTitle(Reviewer.kt:2)
09-18 23:51:10.272 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.AbstractFlashcardViewer.updateDeckName(AbstractFlashcardViewer.kt:3)
09-18 23:51:10.276 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.AbstractFlashcardViewer.updateDeckName(AbstractFlashcardViewer.kt:3)
09-18 23:51:10.277 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.Reviewer.updateScreenCounts(Reviewer.kt:4)
09-18 23:51:10.278 W/AnkiDroid(16934): Backend$checkMainThreadOp/ Op on UI thread: com.ichi2.anki.Reviewer.updateScreenCounts(Reviewer.kt:13)
--------- beginning of crash
```

###### Research
*Enter an [x] character to confirm the points below:*

- [ ] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [ ] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [ ] I have searched for similar existing issues here and on the user forum
- [ ] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by dae**
Resuming the app would trigger a redraw of the counts, and this probably coincided with a card being automatically answered.

Closes #12453

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/anki/CardBrowserTest.kt [hunk 1]
```diff
@@ -554,8 +554,7 @@ class CardBrowserTest : RobolectricTest() {
         // Make sure card has default value in sortType field
         assertThat("Initially Card Browser has order = noteFld", col.get_config_string("sortType"), equalTo("noteFld"))
 
-        // Store the current (before changing the database) Mod Time
-        val initialMod = col.mod
+        col.db.execute("update col set mod = 0")
```
Comment (mikehardy) [inline AnkiDroid/src/test/java/com/ichi2/anki/CardBrowserTest.kt:557]: No philosophical problem with the pragmatism, however, what's the future state look like for this code?
My goal for the question is to have an annotation in here indicating it needs cleanup, ideally with a note (or comment) indicating "Hey future dev, when we're all on the new schema, you'll want to do this instead..."

File: AnkiDroid/src/test/java/com/ichi2/anki/CardBrowserTest.kt [hunk 2]
```diff
@@ -568,9 +567,10 @@ class CardBrowserTest : RobolectricTest() {
         saveControllerForCleanup(cardBrowserController)
 
         // Find the current (after database has been changed) Mod time
-        val finalMod = col.mod
+        col.save()
+        val updatedMod = col.db.queryScalar("select mod from col")
```
Comment (mikehardy) [inline AnkiDroid/src/test/java/com/ichi2/anki/CardBrowserTest.kt:571]: same hope here, for annotation with comment for future dev on what the future-state should look like with new schema only

**Review by mikehardy — CHANGES_REQUESTED** [review]
Thanks a bunch for all of these, the runTest{} change should be a big help as the false-negative rate right now is awful for S/N

Just a couple comments hoping for sage advice in form of annotations on one of the fixes

File: AnkiDroid/src/test/java/com/ichi2/anki/CardBrowserTest.kt (hunk 1 above)
Comment (dae) [reply AnkiDroid/src/test/java/com/ichi2/anki/CardBrowserTest.kt:559]: I've added a RustCleanup annotation (though in the grand scheme of things, it's not a big deal if the test is left as it is)

**Review by mikehardy — APPROVED** [review]
Great - looks good to me, thanks again

**Pull request merged**